File: LeetCode Problems/101 to 200/129. Sum Root to Leaf Numbers/129. Sum Root to Leaf Numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int ans = 0;
    private void traversal(TreeNode root, int pathVal) {
        if (root == null) return;
        pathVal = pathVal * 10 + root.val;
        if (root.left == null && root.right == null) {
            ans += pathVal;
            return;
        }
        traversal(root.left, pathVal);
        traversal(root.right, pathVal);
    }
    public int sumNumbers(TreeNode root) {
        traversal(root, 0);
        return ans;
    }
}

File: LeetCode Problems/101 to 200/129. Sum Root to Leaf Numbers/129. Sum Root to Leaf Numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    int ans = 0;

public:
    void help(TreeNode *root, int val) {
        if (root == nullptr) return;
        val = val * 10 + root->val;
        if (!root->left && !root->right) {
            ans += val;
            return;
        }
        help(root->left, val);
        help(root->right, val);
    }
    int sumNumbers(TreeNode *root) {
        help(root, 0);
        return ans;
    }
};
